utils/catalog_production/add_om10_properties.py

```diff
@@ -0,0 +1,383 @@
+        log_m_err = np.append(log_m_err, float(line[8]))
+    
+    #Build final arrays
+    X = np.vstack([z, v_disp, r_eff, log_m]).T
+    Xerr = np.zeros(X.shape + X.shape[-1:])
+    diag = np.arange(X.shape[-1])
+    
+    Xerr[:, diag, diag] = np.vstack([z_err**2, v_disp_err**2,
+                                    r_eff_err**2, log_m_err**2]).T
+    
+    return X, Xerr
+
+
+#Write new conditioning function
+def get_log_m(cond_indices, m_index, X, model_file, Xerr=None):
+    
+    """
+        Uses a subset of parameters in the given data to condition the
+        model and return a sample value for log(M/M_sun).
+
+        Parameters
+        ----------
+        cond_indices: array_like
+            Array of indices indicating which parameters to use to
+            condition the model. 
+        m_index: int
+            Index of log(M/M_sun) in the list of parameters that were used
+            to fit the model.
+        X: array_like, shape = (n < n_features,)
+            Input data.
+        Xerr: array_like, shape = (X.shape,) (optional)
+            Error on input data. If none, no error used to condition.
+
+        Returns
+        -------
+        log_m: float
+            Sample value of log(M/M_sun) taken from the conditioned model.
+
+        Notes
+        -----
+        The fit_params array specifies a list of indices to use to
+        condition the model. The model will be conditioned and then
+        a mass will be drawn from the conditioned model.
+
+        This is so that the mass can be used to find cosmoDC2 galaxies
+        to act as hosts for OM10 systems.
+
+        This does not make assumptions about what parameters are being
+        used in the model, but does assume that the model has been
+        fit already.
+    """
+
+    if m_index in cond_indices:
+        raise ValueError("Cannot condition model on log(M/M_sun).")
+
+    cond_data = np.array([])
+    if Xerr is not None: cond_err = np.array([])
+    m_cond_idx = m_index
+    n_features = empiricist.XDGMM.mu.shape[1]
+    j = 0
+
+    for i in range(n_features):
+        if i in cond_indices:
+            cond_data = np.append(cond_data,X[j])
+            if Xerr is not None: cond_err = np.append(cond_err, Xerr[j])
+            j += 1
+            if i < m_index: m_cond_idx -= 1
+        else:
+            cond_data = np.append(cond_data,np.nan)
+            if Xerr is not None: cond_err = np.append(cond_err, 0.0)
+
+    if Xerr is not None:
+        cond_XDGMM = empiricist.XDGMM.condition(cond_data, cond_err)
+    else: cond_XDGMM = empiricist.XDGMM.condition(cond_data)
+
+    sample = cond_XDGMM.sample()
+    log_m = sample[0][m_cond_idx]
+    return log_m
+
+
+def estimate_stellar_masses_om10():
+
+    # Instantiate an empiriciSN worker object:
+    empiricist = empiriciSN.Empiricist()
+    X, Xerr = get_sl2s_data()
+
+    # Load in cached om10 catalog
+    hdulist = fits.open('../../data/om10_qso_mock.fits')
```

Comment:
This path is no longer correct (because we are only one directory removed from `$TWINKLES_DIR` in this script).

It is probably safer to use
```
os.path.join(os.environ['TWINKLES_DIR'], 'data', 'om10_qso_mock.fits')
```
which is not dependent on where you are when you run this script.